tests/integration/high_availability/test_replication.py

```diff
@@ -0,0 +1,251 @@
+    primary = await get_primary_unit(ops_test, mysql_unit, mysql_application_name)
+    primary_address = await get_unit_address(ops_test, primary.name)
+
+    # insert some data into the mysql cluster
+    database_name, table_name = "test-check-consistency", "data"
+    inserted_value = generate_random_string(255)
+    server_config_credentials = await get_server_config_credentials(primary)
+    insert_data_sql = [
+        f"CREATE DATABASE IF NOT EXISTS `{database_name}`",
+        f"CREATE TABLE IF NOT EXISTS `{database_name}`.`{table_name}` (id varchar(255), primary key(id))",
+        f"INSERT INTO `{database_name}`.`{table_name}` (id) VALUES ('{inserted_value}')",
+    ]
+
+    await execute_queries_on_unit(
+        primary_address,
+        server_config_credentials["username"],
+        server_config_credentials["password"],
+        insert_data_sql,
+        commit=True,
+    )
+
+    # ensure that the inserted data gets replicated onto every instance in the mysql cluster
+    select_inserted_data_sql = [
+        f"SELECT id FROM `{database_name}`.`{table_name}` WHERE id = '{inserted_value}'",
+    ]
+
+    for unit in ops_test.model.applications[mysql_application_name].units:
+        unit_address = await get_unit_address(ops_test, unit.name)
+
+        try:
+            for attempt in Retrying(stop=stop_after_delay(10), wait=wait_fixed(2)):
+                with attempt:
+                    output = await execute_queries_on_unit(
+                        unit_address,
+                        server_config_credentials["username"],
+                        server_config_credentials["password"],
+                        select_inserted_data_sql,
+                    )
+                    assert output[0] == inserted_value
+        except RetryError:
+            assert False, f"Unable to query inserted data from unit {unit.name}"
+
+
+@pytest.mark.order(3)
+@pytest.mark.abort_on_fail
+@pytest.mark.replication_tests
+async def test_no_replication_across_clusters(ops_test: OpsTest) -> None:
+    """Test to ensure that writes to one cluster do not replicate to another cluster."""
+    mysql_application_name, application_name = await high_availability_test_setup(ops_test)
+    application_unit = ops_test.model.applications[application_name].units[0]
+
+    # assert that there are 3 units in the mysql cluster
+    assert len(ops_test.model.applications[mysql_application_name].units) == 3
+
+    # stop the continuous writes for this test
+    clear_writes_action = await application_unit.run_action("clear-continuous-writes")
+    await clear_writes_action.wait()
+
+    # deploy another mysql application cluster with the same 'cluster-name'
+    another_mysql_application_name = "another"
+    await deploy_and_scale_mysql(
+        ops_test,
+        check_for_existing_application=False,
+        mysql_application_name=another_mysql_application_name,
+    )
+
+    # insert some data into the mysql cluster
```

Comment:
```suggestion
    # insert some data into the first mysql cluster
```

Reply:
addressed in ef4d0f84c9105de5cec89b0c2468f1513e867df3